### ISSUE TITLE
Add XGBoost classifier, regressor (and ranker

### ISSUE THREAD

**Issue by bgraham89** [bgraham89/Machine-Learning#2]
XGBoost is a powerful ensemble method, based on decision trees and gradient boosting, that frequently outperforms other machine learning models. (It's also relatively easy to implement, with a scikit-learn interface)

**Comment by bgraham89** [comment]
- [ ] XGBregressor
- [ ] XGBclassifier
- [ ] XGBranker
